scripts/linters/pylint_extensions.py

```diff
@@ -2919,6 +2919,85 @@ def visit_importfrom(self, node: astroid.nodes.ImportFrom) -> None:
                 self.add_message('disallowed-text-import', node=node)
 
 
+# TODO(#16567): Here we use MyPy ignore because of the incomplete typing of
+# pylint library and absences of stubs in pylint, forces MyPy to
+# assume that BaseChecker class has attributes of type Any.
+# Thus to avoid MyPy's error
+# (Class cannot subclass 'BaseChecker' (has type 'Any')),
+# we added an ignore here.
+class NoBlankLineAfterFunctionDefChecker(checkers.BaseChecker):  # type: ignore[misc]
+    """Ensures that there is no blank line after function definition"""
+
+    __implements__ = (interfaces.IAstroidChecker,)
+
+    name = 'no-blank-line'
+    msgs = {
+        'C0041': (
+            'Blank line after function definition',
+            'Blank-line-after-function-definition',
+            'Please remove blank line after function definition',
+        )
+    }
+
+    def visit_functiondef(self, node: astroid.nodes.FunctionDef) -> None:
+        """Called for function and method definitions (def).
+
+        Args:
+            node: astroid.scoped_nodes.FunctionDef. Node for a function or
+                method definition in the AST.
+        """
+        if node.doc:
```

Comment:
Please add comments describing what each `if` condition is checking and what it means to enter the `if` or `else` block.